source/rust_verify/src/rust_intrinsics_to_vir.rs

```diff
@@ -0,0 +1,80 @@
+use crate::util::spanned_typed_new;
+use rustc_span::Span;
+use std::sync::Arc;
+use vir::ast::{Expr, ExprX, IntRange, IntegerTypeBoundKind, Mode, Typ, TypX, UnaryOpr};
+
+pub(crate) fn int_intrinsic_constant_to_vir(span: Span, typ: &Typ, f_name: &str) -> Option<Expr> {
+    let mk_expr = |x: ExprX| spanned_typed_new(span, &typ, x);
+
+    let lit_expr = |u: u128| Some(mk_expr(ExprX::Const(vir::ast_util::const_int_from_u128(u))));
+
+    let lit_expr_i = |i: i128| Some(mk_expr(ExprX::Const(vir::ast_util::const_int_from_i128(i))));
+
+    let arch_bits = || {
+        let arg = spanned_typed_new(
+            span,
+            &Arc::new(TypX::Int(IntRange::Int)),
+            ExprX::Const(vir::ast_util::const_int_from_u128(0)),
+        );
+
+        let kind = IntegerTypeBoundKind::ArchWordBits;
+
+        return mk_expr(ExprX::UnaryOpr(UnaryOpr::IntegerTypeBound(kind, Mode::Exec), arg));
+    };
+
+    let arch_bound = |kind| {
+        return Some(mk_expr(ExprX::UnaryOpr(
+            UnaryOpr::IntegerTypeBound(kind, Mode::Exec),
+            arch_bits(),
+        )));
+    };
+
+    match f_name {
```

Comment:
Is there a way to get the `DefId`s for these, instead of matching the string names?
(I know we do this elsewhere, but I'd rather we didn't depend on how these are printed.)

Reply:
So, it turns out they're defined not as intrinsics, but with this macro:

https://github.com/rust-lang/rust/blob/master/library/core/src/num/int_macros.rs#L18

```rust
pub const MIN: Self = !0 ^ ((!0 as $UnsignedT) >> 1) as Self;
```

So: nevermind, I guess? 😬

Reply:
I was wondering whether it would make sense to have these in pervasive instead, with a specification that ties them to the intrinsic (we've done something similar for strings). However, I don't think that would work directly, because you'd want the actual constant from `core`.

What about something like:
```rust
#[verifier(external_body)]
pub const fn i64_max() -> (v: i64)
    ensures v == signed_max(64) {
    i64::MAX
}
```
(maybe properly grouped for each type)

My reasoning here is that these should not need a special code path once we support specification for `core`/`std`: their specification can just refer to the intrinsic.
I'm not necessarily asking for this change, just bringing it up as an option.